repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala

```diff
@@ -55,518 +54,373 @@ class KeystoneV2Filter @Inject()(configurationService: ConfigurationService,
-          } else {
-            val processingResult =
-              getAuthToken flatMap { authToken =>
-                validateToken(authToken) flatMap { validToken =>
-                  val tenantScopedRoles = getTenantScopedRoles(validToken.roles)
-                  val userIsPreAuthed = isUserPreAuthed(tenantScopedRoles)
-                  val scopedRolesToken = if (userIsPreAuthed) validToken else validToken.copy(roles = tenantScopedRoles)
-                  val doTenantCheck = shouldAuthorizeTenant(userIsPreAuthed)
-                  val matchedUriTenant = getMatchingUriTenant(doTenantCheck, scopedRolesToken)
-                  addTokenHeaders(scopedRolesToken, matchedUriTenant)
-                  authorizeTenant(doTenantCheck, matchedUriTenant) flatMap { _ =>
-                    lazy val endpoints = getEndpoints(authToken) // Prevents making call if its not needed
-                    addCatalogHeader(endpoints) flatMap { _ =>
-                      authorizeEndpoints(userIsPreAuthed, endpoints) flatMap { _ =>
-                        addGroupsHeader(getGroups(authToken, scopedRolesToken))
-                      }
-                    }
-                  }
-                }
-              }
 
-            processingResult match {
-              case Success(_) => Pass
-              case Failure(e: MissingAuthTokenException) => Reject(SC_UNAUTHORIZED, Some(e.getMessage))
-              case Failure(e: NotFoundException) => Reject(SC_UNAUTHORIZED, Some(e.getMessage))
-              case Failure(e: InvalidTenantException) => Reject(SC_UNAUTHORIZED, Some(e.getMessage))
-              case Failure(e: UnparseableTenantException) => Reject(SC_UNAUTHORIZED, Some(e.getMessage))
-              case Failure(e: IdentityCommunicationException) => Reject(SC_BAD_GATEWAY, Some(e.getMessage))
-              case Failure(e: UnauthorizedEndpointException) => Reject(SC_FORBIDDEN, Some(e.getMessage))
-              case Failure(e: OverLimitException) =>
-                response.addHeader(HttpHeaders.RETRY_AFTER, e.retryAfter)
-                if (isSelfValidating) {
-                  Reject(e.statusCode, Some(e.getMessage))
-                } else {
-                  Reject(SC_SERVICE_UNAVAILABLE, Some(e.getMessage))
-                }
-              case Failure(e) if e.getCause.isInstanceOf[AkkaServiceClientException] && e.getCause.getCause.isInstanceOf[TimeoutException] =>
-                Reject(SC_GATEWAY_TIMEOUT, Some(s"Call timed out: ${e.getMessage}"))
-              case Failure(_: AdminTokenUnauthorizedException) if isSelfValidating =>
-                Reject(SC_UNAUTHORIZED, Some("Token unauthorized"))
-              case Failure(e) => Reject(SC_INTERNAL_SERVER_ERROR, Some(e.getMessage))
-            }
-          }
+  override def doWork(servletRequest: HttpServletRequest, servletResponse: HttpServletResponse, chain: FilterChain): Unit = {
+    val response = new HttpServletResponseWrapper(servletResponse, MUTABLE, PASSTHROUGH)
```

Comment:
I don't love having to wrap the response here to then wrap it again when `super.doWork` is called. I get that we do that so that we can keep the super code common yet still `uncommit` and fix the things that need to be fixed, but still, bleh. 

I guess the alternative would be to add a hook in the abstract filter to allow processing of the response before it is committed, but then the wrapper response would have to be passed if we wanted to avoid re-wrapping. I'm not sure if that's actually any better.

Reply:
Yep, another 'fix' i wasn't super happy with, but had to live with.